packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -657,6 +684,114 @@ describe('CDK Include', () => {
       cfnTemplate.getOutput('FakeOutput');
     }).toThrow(/Output with logical ID 'FakeOutput' was not found in the template/);
   });
+
+  test('replaces references to parameters with the user-specified values in Resources, Conditions, Metadata, and Options', () => {
+    new inc.CfnInclude(stack, 'includeTemplate', {
+      templateFile: _testTemplateFilePath('parameter-references.json'),
+      parameters: {
+        'MyParam': 'my-s3-bucket',
+      },
+    });
+
+    expect(stack).toMatchTemplate({
+      "Transform": {
+        "Name": "AWS::Include",
+        "Parameters": {
+          "Location": "my-s3-bucket",
+        },
+      },
+      "Metadata": {
+        "Field": {
+          "Fn::If": [
+            "AlwaysFalse",
+            "AWS::NoValue",
+            "my-s3-bucket",
+          ],
+        },
+      },
+      "Conditions": {
+        "AlwaysFalse": {
+          "Fn::Equals": [ "my-s3-bucket", "Invalid?BucketName"],
+        },
+      },
+      "Resources": {
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Metadata": {
+            "Field": "my-s3-bucket",
+          },
+          "Properties": {
+            "BucketName": "my-s3-bucket",
+          },
+        },
+      },
+      "Outputs": {
+        "MyOutput": {
+          "Value": "my-s3-bucket",
+        },
+      },
+    });
+  });
+
+  test('can replace parameters in Fn::Sub', () => {
+    new inc.CfnInclude(stack, 'includeTemplate', {
+      templateFile: _testTemplateFilePath('fn-sub-parameters.json'),
```

Comment:
Do you just want to add `parameters` as a property to `IncludeTestTemplateProps`? This way, you'll be able to still use the `includeTestTemplate` helper.

Reply:
this is a much better way of doing it, thanks.